src/config.c

```diff
@@ -225,21 +228,70 @@ void config_vifs_from_kernel(void)
     struct uvif *v;
     vifi_t vifi;
     uint32_t addr, mask, subnet;
-    struct ifaddrs *ifaddr, *ifa;
+    struct ifaddrs *ifaddr, *ifa,*ifap;
+    int phyint_num;
+    int count;
+    struct iflist *entry;
 
     /* Query config first for list of enabled interfaces */
-    build_iflist();
+    phyint_num = build_iflist();
 
+init_vif_list:
     total_interfaces = 0; /* The total number of physical interfaces */
     if (getifaddrs(&ifaddr) == -1) {
 	logit(LOG_ERR, errno, "Failed retrieving interface addresses");
 	return;
     }
 
+    ifap = calloc(phyint_num, (sizeof(struct ifaddrs)));
+    if(!ifap) {
+	    logit(LOG_ERR,0,"%s[%d]: Allocation error",__func__,__LINE__);
+	    return;
+    }
+
+    for (count=0,ifa = ifaddr; ifa; ifa = ifa->ifa_next) {
+	LIST_FOREACH(entry, &il, link) {
+//		logit(LOG_INFO,0,"name %s flag %x ifa_addr %p",ifa->ifa_name,ifa->ifa_flags,ifa->ifa_addr);
+
+	    if(((ifa->ifa_flags & IFF_UP) != IFF_UP) || strcmp(entry->ifname,ifa->ifa_name))
```

Comment:
Again, lots of missing spaces.  Inconsistent also with your own style in the following if() statement.